spec/models/concerns/smart_filterable_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe SmartFilterable do
+  before(:all) do
+    FactoryGirl.create(:comment)
+    FactoryGirl.create(:post)
+  end
+
+  it "can filter string correctly" do
+    result = Comment.smart_filter(content: "万")
+    expect(result).to eq(Comment.all.where('content LIKE ?', '%万%'))
+  end
+
+  it "can filter datetime correctly" do
+    result = Post.smart_filter(created_at: "today")
+    expect(result).to eq(Post.all.where(created_at: \
+      Date.today.beginning_of_day..Date.today.end_of_day))
```

Comment:
I will write it as:

```ruby
within_today = Date.today.beginning_of_day..Date.today.end_of_day
todays_posts = Post.where(created_at: within_today)
expect(result.pluck(:id)).to match_array(todays_posts.pluck(:id))
```

1. extract long fragments into named variables to make them self-documented
2. use `match_array` instead of `eq` when we don't care about the order
3. by compare only on `id` we ignore the comparison on attributes explicitly, and this is presumably faster